packages/kbn-unified-data-table/src/components/custom_control_columns/color_indicator/color_indicator_control_column.tsx

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useContext } from 'react';
+import { css } from '@emotion/react';
+import { EuiDataGridControlColumn, useEuiTheme, EuiThemeComputed } from '@elastic/eui';
+import type { DataTableRecord } from '@kbn/discover-utils';
+import { UnifiedDataTableContext } from '../../../table_context';
+
+const COLOR_INDICATOR_WIDTH = 4;
+
+interface ColorIndicatorCellParams {
+  rowIndex: number;
+  getRowIndicatorColor: (row: DataTableRecord, euiTheme: EuiThemeComputed) => string | undefined;
+}
+
+const ColorIndicatorCell: React.FC<ColorIndicatorCellParams> = ({
+  rowIndex,
+  getRowIndicatorColor,
+}) => {
+  const { euiTheme } = useEuiTheme();
+  const { rows } = useContext(UnifiedDataTableContext);
+  const row = rows[rowIndex];
+  const color: string = (row && getRowIndicatorColor(row, euiTheme)) || 'transparent';
+  return (
+    <div
+      data-test-subj="unifiedDataTableRowColorIndicatorCell"
+      css={css`
+        background: ${color};
+        width: ${COLOR_INDICATOR_WIDTH}px;
+        height: 100%;
+      `}
```

Comment:
it looks and works great, one ask I have. We now have a control column without functionality, that's the only disadvantage of this approach (and it makes sense to implement it like this). 

I think we should add a title / aria label describing which log level is represented by this element.

Reply:
@kertal It can become too noisy/distracting if we add such labels for a screenreader. I might be wrong but I think it's better to omit it in this case.

Reply:
Let's ask @1Copenut about that. this is an element indicating the severity of given a given log level. It looks like this. The component being used is EuiDataGrid, what's best practice for screenreaders in this case:
![Discover_-_Elastic](https://github.com/elastic/kibana/assets/463851/2c23deab-b542-400e-bce4-07035440f024)

Reply:
Thank you @kertal for the link to review. I took a look and confirmed the log level cell will take keyboard focus, as is by design for `EuiDataGrid`. Given that fact, I think we should have an accessible label. Looking into the DOM a bit more, I found an `EuiScreenReaderOnly` with a hidden P tag. We could update it something like this:

```javascript
<EuiScreenReaderOnly>
  <p>Row 1, alert</p>
</EuiScreenReaderOnly>
```

By removing the `hidden` attribute and making the SR-only message relate to the row and log type, we give screen reader users a text representation of what the color is indicating.

Reply:
Thanks, @1Copenut!

The `p` tag you see in the DOM is added by Eui and we can't append directly to it. I added `title` attribute to the indicator tag instead and Voice Over reads it correctly as you suggested https://github.com/elastic/kibana/pull/186418/commits/b903ba1347dc4a81e28902a34bacbf7c24a9c997